raster/src/main/scala/geotrellis/raster/imagery/CloudRemoval.scala

```diff
@@ -0,0 +1,57 @@
+package geotrellis.raster.imagery
+
+import geotrellis.raster._
+import geotrellis.raster.render._
+import geotrellis.raster.io.geotiff.SingleBandGeoTiff
+
+import java.io.File
+import spire.syntax.cfor._
+
+object CloudRemoval {
+
+  def cloudRemovalSingleBand(images: Array[Tile], threshold: Int) : Tile = {
+    val headImage = images(0)
+    val result = ArrayTile.empty(headImage.cellType, headImage.cols, headImage.rows)
+
+    cfor(0)(_ < result.rows, _ + 1) { row =>
+      cfor(0)(_ < result.cols, _ + 1) { col =>
+        var sum = 0
+        var count = 0
+        cfor(0)(_ < images.length, _ + 1) { i =>
+          val v = images(i).get(col, row)
+          if(isData(v) && v < threshold) {
+            sum += v
+            count += 1
+          }
+        }
+        
+        result.set(col, row, sum / count)
```

Comment:
Ya, there's no specific reason for choosing mean over median. I used mean because folks at Mapbox had done a similar averaging for their cloudless atlas. Mode is not used because the difference in frequency of pixel intensities is hardly 1 or 2 in 32/16 bit images (which I used for testing). I am yet to test this using median. Could give it a shot. But I doubt its importance in the output image.